Fix memory leak in `editable_gesture_test.dart`

ValentinVignal opened the pull request:
Part of https://github.com/flutter/flutter/issues/141198

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

ValentinVignal committed:
test: Fix leaks in editable gesture test

ValentinVignal commented:
cc @polina-c

ValentinVignal commented on packages/flutter/test/rendering/editable_gesture_test.dart:
```diff
@@ -11,18 +11,17 @@ import 'package:leak_tracker_flutter_testing/leak_tracker_flutter_testing.dart';
     expect(WidgetsBinding.instance, binding);
     final TextSelectionDelegate delegate = FakeEditableTextState();
+    final ViewportOffset offset = ViewportOffset.zero();
+    addTearDown(offset.dispose);
     final RenderEditable editable = RenderEditable(
       backgroundCursorColor: Colors.grey,
       selectionColor: Colors.black,
       textDirection: TextDirection.ltr,
       cursorColor: Colors.red,
-      offset: ViewportOffset.zero(),
+      offset: offset,
```
I'm not too sure about this one, should `editable.dispose` dispose its `offset` ?

polina-c replied:
Nope, editable did not create offset, so it should not dispose it. So, your way to do it is correct.

polina-c commented:
Looks good.
Did you enable leak tracker, repro the failure and made sure your fix resolves it?

ValentinVignal commented:
> Did you enable leak tracker, repro the failure and made sure your fix resolves it?

@polina-c I changed 

```diff
-  if (_isLeakTrackingEnabled()) {
+ if (true) {
```

into `packages/flutter/test/flutter_test_config.dart` and the test was failing when I removed the `experimentalLeakTesting: LeakTesting.settings.withIgnoredAll(),`. After my changes it passes.

Is that the correct way to do it?

polina-c commented:
> Is that the correct way to do it?

Yes, thank you!!!
